examples/systemcalls/systemcalls.c

```diff
@@ -40,60 +55,106 @@ bool do_exec(int count, ...)
     va_start(args, count);
     char * command[count+1];
     int i;
+    int status;
     for(i=0; i<count; i++)
     {
         command[i] = va_arg(args, char *);
     }
     command[count] = NULL;
-    // this line is to avoid a compile warning before your implementation is complete
-    // and may be removed
-    command[count] = command[count];
-
-/*
- * TODO:
- *   Execute a system command by calling fork, execv(),
- *   and wait instead of system (see LSP page 161).
- *   Use the command[0] as the full path to the command to execute
- *   (first argument to execv), and use the remaining arguments
- *   as second argument to the execv() command.
- *
-*/
 
+    fflush(stdout);
+    pid_t pid = fork();
+    if (pid == 0) {
+        execv(command[0], command);
+        perror("Command execution failed.");
+        va_end(args);
+        exit(-1);
+    }
+    
+    waitpid(pid, &status, 0);
+    if (WIFEXITED(status)) {
+        if (WEXITSTATUS(status) == 0) {
+            printf("Child execution successful.\n");
+            va_end(args);
+            return true;
+        } else {
+            perror("Child execution failed.");
+            va_end(args);
+            return false;
+        }
+    } else {
+        perror("Child exited abnormally.");
+        va_end(args);
+        return false;
+    }
     va_end(args);
-
-    return true;
+    return false;
 }
 
 /**
 * @param outputfile - The full path to the file to write with command output.
 *   This file will be closed at completion of the function call.
 * All other parameters, see do_exec above
+*
+* Referenced https://stackoverflow.com/questions/13784269/redirection-inside
+* -call-to-execvp-not-working/13784315#13784315
 */
 bool do_exec_redirect(const char *outputfile, int count, ...)
 {
     va_list args;
     va_start(args, count);
     char * command[count+1];
     int i;
+    int status;
     for(i=0; i<count; i++)
     {
         command[i] = va_arg(args, char *);
     }
     command[count] = NULL;
-    // this line is to avoid a compile warning before your implementation is complete
-    // and may be removed
-    command[count] = command[count];
-
-
-/*
- * TODO
- *   Call execv, but first using https://stackoverflow.com/a/13784315/1446624 as a refernce,
- *   redirect standard out to a file specified by outputfile.
- *   The rest of the behaviour is same as do_exec()
- *
-*/
 
+    int fd = creat(outputfile, 0644);
```

Comment:
@tiba6275 think about how you could avoid duplication between these two functions.